### PR TITLE
+animation support (parser rewrite)

### PR DESCRIPTION
**Pull request by ranfdev**
I've rewritten the parser to support keyframes. Everything else works as expected.
example: 
```js
const Container = ps('div')({
  animation: 'zoom 1s',
  '@keyframes zoom': {
    from: {
      transform: 'scale(0.5)'
    },
    to: {
      transform: 'scale(2)'
    }
  }
})
```

### CODE REVIEW

**Comment by ranfdev** [comment]
This pr should fix #36 and #27

**Comment by morishitter** [comment]
@ranfdev Thank you for your work!

> Should i write a utility function to create re-usable keyframes?

I think, it seems better than the current implementation because we can reuse `zoom` in other components.

**Comment by ranfdev** [comment]
Now you can do this: 
```js
import {keyframes} from 'picostyle'
const zoom = keyframes({
    from: {
      transform: 'scale(0.5)'
    },
    to: {
      transform: 'scale(2)'
    }
})
const Container = ps('div')({
  animation: `${zoom} 300ms`,
})
```

**Comment by morishitter** [comment]
@ranfdev Awesome!! Could you add the unit test for `keyframes` function, too?

**Comment by ranfdev** [comment]
Ok, i will!

**Comment by morishitter** [comment]
Please run `yarn run format`.

**Comment by ranfdev** [comment]
All should work fine.
With this new parser, every ```&``` is replaced with the classname, so you can use every selector you want.
Please, view the test file and the ```1 level descendant combinator``` test. There's a little **important** change. Now if you want to select every ```p``` inside your component, you have to do "& p"

File: test/basic.test.js [hunk 1]
```diff
@@ -49,23 +49,25 @@ test("basic style object", () => {
...
   })
   expectClassNameAndCssText(
     Test(),
     "p2",
-    ".p2 {color: red;},.p2 tr {color: white;}"
+    ".p2 tr {color: white;},.p2 {color: red;}"
   )
 })
 
+// This works in the browser, but not here. I don't know why
```
Comment (morishitter) [inline test/basic.test.js]: What is the version of Node.js in your environment? It passed in TravisCI.

**Comment by morishitter** [comment]
@ranfdev Awesome work! 

> Now if you want to select every p inside your component, you have to do "& p"

I agree with that. It seems nice.

File: test/basic.test.js (hunk 1 above)
Comment (ranfdev) [reply test/basic.test.js]: My version is: v9.11.1. TravisCI doesn't run ```npm run test```, it only runs ```lint```,```tslint``` and ```size-check```

**Comment by ranfdev** [comment]
I had to refactor the code for the new "& ..." syntax. It's now fixed!

**Comment by morishitter** [comment]
@ranfdev Great! Thank you for your work. I will release v2.0.0 including this change.

**Pull request merged**